### PR TITLE
Add integration tests for tags plus fix permissions

### LINKED ISSUE #22512: Regular users can create invisible or non-"userAssignable" tags
`/remote.php/dav/systemtags/` allows regular users to create invisible or non-"userAssignable" tags. I assume this behaviour should be reserved for admins?

```http
POST /remote.php/dav/systemtags/ HTTP/1.1
Host: localhost
User-Agent: Mozilla/5.0 (Macintosh; Intel Mac OS X 10.11; rv:46.0) Gecko/20100101 Firefox/46.0
Accept: text/html,application/xhtml+xml,application/xml;q=0.9,*/*;q=0.8
Accept-Language: en-US,en;q=0.5
Accept-Encoding: gzip, deflate
X-Requested-With: XMLHttpRequest
Content-Type: application/json
Content-Length: 70
Cookie: oc3ncedmobud=toouiju9jl90tpjk06c50grhn1; oc_sessionPassphrase=T%2FEvYm%2BBBWw9TctnL8qxbt5h77NJjmjBFKGBiRsagKT6FMzz9qNoBrrQTyQVukiHEfljuuxlcAzYy0xQDxi8rV1kQksPL0SML%2F35Ww4O21hyj9tGvzP0oklR52DplzKp; ocjcd5zbuzxl=f5stbvesjrnmsibbd7ke635jc6; och9myy6w5uq=ofa4n30ip1fend7eqvqoaff506; ocn9qnveszb6=gkspstac6vrdvtckedo9ike7d3; ocsdfaof31in=584u4s1u5ge8bispi1lprcei56; oc9ytv0hw06z=rrkadaeu0l4tf3750oalaveha4
Connection: close

{"name":"MySuperHiddenTag","userVisible":false,"userAssignable":false}
```

cc @PVince81 @karlitschek Expected behaviour or bug?

### PR DESCRIPTION
**Pull request by LukasReschke**
Adds a bunch of mainly permission related integration tests.

Fixes #22512

### CODE REVIEW

**Comment by LukasReschke** [comment]
@PVince81 Mind taking a look? THX.

**Comment by LukasReschke** [comment]
Also @SergioBertolinSG :)

File: apps/dav/lib/server.php [hunk 1]
```diff
@@ -93,7 +93,11 @@ public function __construct(IRequest $request, $baseUri) {
 		$this->server->addPlugin(new \OCA\DAV\CardDAV\Plugin());
 
 		// system tags plugins
-		$this->server->addPlugin(new \OCA\DAV\SystemTag\SystemTagPlugin(\OC::$server->getSystemTagManager()));
+		$this->server->addPlugin(new \OCA\DAV\SystemTag\SystemTagPlugin(
+			\OC::$server->getSystemTagManager(),
+			\OC::$server->getGroupManager(),
+			\OC::$server->getUserSession()
```
Comment (rullzer) [inline apps/dav/lib/server.php]: GroupManager and UserSession seems to be the wrong way around.j

File: apps/dav/lib/systemtag/systemtagplugin.php [hunk 2]
```diff
@@ -61,12 +63,26 @@ class SystemTagPlugin extends \Sabre\DAV\ServerPlugin {
...
+
+	/**
 	 * @param ISystemTagManager $tagManager tag manager
+	 * @param IGroupManager $groupManager
+	 * @param IUserSession $userSession
 	 */
-	public function __construct(ISystemTagManager $tagManager) {
+	public function __construct(ISystemTagManager $tagManager,
+								IGroupManager $groupManager,
+								IUserSession $userSession) {
```
Comment (rullzer) [inline apps/dav/lib/systemtag/systemtagplugin.php]: See here

**Comment by rullzer** [comment]
:rocket: :+1:

**Comment by MorrisJobke** [comment]
:roller_coaster: :+1:

**Comment by nickvergessen** [comment]
Actual code change is good :+1:

**Pull request merged**